publish.js

```diff
@@ -258,22 +258,41 @@ publish.tagVC = function (isTest, version, options) {
 };
 
 /**
- * Restore the package.json file to the latest committed version.
+ * Restore the files specified in filesToClean option to the latest committed version.
+ * Restores package.json and package-lock.json files to the latest commited version by default.
  *
- * Used internally to reset version number changes in package.json
- * @param moduleRoot {String} - the directory where the package.json file to
+ * Used internally to reset version number changes in list of files in filesToClean option.
+ * First checks whether the list of files in filesToClean option are being tracked by git or not.
+ * @param moduleRoot {String} - the directory where the files to 
                                 clean is located in.
- * @param options {Object} - e.g. {"cleanCmd": "git checkout -- package.json"}
+ * @param options {Object} - e.g. {"checkFilesTracking": "git ls-files --error-unmatch ${filesToClean}", "cleanCmd": "git checkout -- ${filesToClean}", "filesToClean": "package.json package-lock.json"}
  */
 publish.clean = function (moduleRoot, options) {
     var originalDir = process.cwd();
 
     // change to the module root directory
     process.chdir(moduleRoot || "./");
 
-    // run the clean command
-    publish.execSyncFromTemplate(options.cleanCmd);
+    var file = options.filesToClean.split(" ");
+    
+    file.forEach(function(file){
+        var isTracked = true;
 
+        // check whether file is being tracked or not
+        try {publish.execSyncFromTemplate(options.checkFilesTracking, {
+                filesToClean: file
+            }, options.checkFilesTrackingHint);
+        } catch (error) {
+            isTracked = false;
+        }
+        
+        if (isTracked)
+            // run the clean command if file is being tracked
+            publish.execSyncFromTemplate(options.cleanCmd, {
+                filesToClean: file
+            });
```

Comment:
You might be able to simplify this by putting this step in the try block as well. Assuming you can differentiate between types of errors.